docs/tutorials/event-driven-microservice.md

```diff
@@ -0,0 +1,613 @@
+
+Before you can begin coding your microservice, you'll need access to the Avro schemas that the {{ site.ak }} topic is serialized with. Confluent has [a Maven plugin](https://docs.confluent.io/current/schema-registry/develop/maven-plugin.html) that makes this simple, which you might have already noticed is present in the `pom.xml` file. Run the following command, which downloads the required Avro schema out of {{ site.sr }} to your local machine:
+
+```
+mvn schema-registry:download
+```
+
+You should now have a file called `src/main/avro/possible_anomalies-value.avsc`. This is the Avro schema generated by ksqlDB for the value portion of the {{ site.ak }} records of the `possible_anomalies` topic.
+
+Next, compile the Avro schema into a Java file. The [Avro Maven plugin](https://avro.apache.org/docs/current/gettingstartedjava.html) (already added to the `pom.xml` file, too) makes this simple:
+
+```
+mvn generate-sources
+```
+
+You should now have a file called `target/generated-sources/io/ksqldb/tutorial/PossibleAnomaly.java` containing the compiled Java code.
+
+### Write the Kafka consumer code
+
+Now we can write the code that triggers side effects when anomalies are found. Add the following Java file at `src/main/java/io/ksqldb/tutorial/EmailSender.java`. This is a simple program that consumes events from {{ site.ak }} and sends an email with SendGrid for each one it finds. There are a few constants to fill in, including a SendGrid API key. You can get one by signing up for SendGrid.
+
+```java
+package io.ksqldb.tutorial;
+
+import org.apache.kafka.clients.consumer.ConsumerConfig;
+import org.apache.kafka.clients.consumer.ConsumerRecord;
+import org.apache.kafka.clients.consumer.ConsumerRecords;
+import org.apache.kafka.clients.consumer.KafkaConsumer;
+import org.apache.kafka.common.serialization.StringDeserializer;
+
+import io.confluent.kafka.serializers.KafkaAvroDeserializer;
+import io.confluent.kafka.serializers.KafkaAvroDeserializerConfig;
+import io.confluent.kafka.serializers.AbstractKafkaAvroSerDeConfig;
+
+import com.sendgrid.SendGrid;
+import com.sendgrid.Request;
+import com.sendgrid.Response;
+import com.sendgrid.Method;
+import com.sendgrid.helpers.mail.Mail;
+import com.sendgrid.helpers.mail.objects.Email;
+import com.sendgrid.helpers.mail.objects.Content;
+
+import java.time.Duration;
+import java.time.Instant;
+import java.time.ZoneId;
+import java.time.format.DateTimeFormatter;
+import java.time.format.FormatStyle;
+import java.util.Collections;
+import java.util.Properties;
+import java.util.Locale;
+import java.io.IOException;
+
+public class EmailSender {
+
+    private final static String BOOTSTRAP_SERVERS = "localhost:29092";
+    private final static String SCHEMA_REGISTRY_URL = "http://localhost:8081";
+    private final static String TOPIC = "possible_anomalies";
+    private final static String FROM_EMAIL = "test@example.com";
+    private final static String SENDGRID_API_KEY = "";
```

Comment:
Maybe add comments to make it clear which of these fields need to be modified/populated by the tutorial user? "There are a few constants to fill in, including a SendGrid API key" is a bit vague otherwise.

Reply:
Good call - I added some more comments.